### ISSUE TITLE
example.js fails after a few seconds with "Cannot establish context"

### ISSUE THREAD

**Issue by CoryGH** [cdesiles/node-pcsc#7]
Here's the output if it helps:

$ sudo node example.js

/home/username/node-pcsc/lib/node-pcsc.js:22
var pcsc =  new PCSC();
            ^
TypeError: Cannot establish context
    at Object.<anonymous> (/home/username/node-pcsc/lib/node-pcsc.js:22:13)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:901:3

**Comment by CoryGH** [comment]
Sorry for this, I left out:

sudo apt-get install pcscd

So I didn't start the process up beforehand, after it has been installed and started it works fine.

You should know that the libpcsclite package doesn't exist (at least for Ubuntu's apt-get command).

**Comment by LudovicRousseau** [comment]
libpcsclite is called libpcsclite1.
Fixed in https://github.com/jokesterfr/node-pcsc/pull/8

**Comment by LudovicRousseau** [comment]
Another point, never run command as root (using sudo), unless you really, really know what you do.

**Comment by CoryGH** [comment]
I've had issues using node-usb as a standard user (and ffmpeg when interacting with /dev/video\* paths) - I assumed root access was required to interact directly with hardware - I know after seeing how this failed for me that pcscd handles the hardware interface and just relays commands, but without reviewing the code I assumed node-pcsc interacted directly with the hardware.

**Comment by cdesiles** [comment]
libpcsclite1 has been set in the apt-get install command, as in #8  
Is it fixed for you now?

**Comment by CoryGH** [comment]
Yes and no.  I still had to do:
sudo apt-get install pcscd

in order to get it connecting up (this time I used Raspbian instead of Ubuntu to make sure I had a clean install - node.js was compiled manually over serveral hours due to some unrelated issues with node-gyp on ARM so I built the source directly from a git clone of the github repo instead of relying upon 'npm install[ -g] node-pcsc').

On a (possibly related note, though this might be a new issue):

The npm source might be out of date (along with example.js) - going through the code for node-pcsc it doesn't appear there is any spot that emits a message called 'message' - however using the code inside of the ".on('message'..." bit of example.js inside of example2.js's ".on('evt'..." block appears to work the same.

TL;DR: should add:
sudo apt-get install pcscd
to README.md for Ubuntu/Raspbian (for Raspbian node.js works best installed via Debian Wheezy instructions found at: https://github.com/joyent/node/wiki/Installing-Node.js-via-package-manager)

And make a new example.js file like:

var pcsc = require('node-pcsc');

pcsc.init();

pcsc.on('evt', function(evt) {
        console.log('Event status:',evt.reader.status);
        // Show info when state switch to present
        if(evt.reader.status == 'SCARD_STATE_PRESENT'){
                console.log(evt.reader.name);
                if (evt.reader.type) console.log('-> type:', evt.reader.type);
                if (evt.card.name) console.log('-> card:', evt.card.name);
                if (evt.card.ATR) console.log('-> card ATR:', evt.card.ATR);
                if (evt.card.Protocol) console.log('-> card Protocol:', evt.card.Protocol);
                if (evt.card.err) console.log('-> card transmit err:', evt.card.err);
                if (evt.card.UID) console.log('-> card UID:', evt.card.UID);
                console.log();
        }
});

console.log('started');
